orttraining/orttraining/test/training_ops/cuda/layer_norm_test.cc

```diff
@@ -97,5 +97,95 @@ TEST(CudaKernelTest, LayerNormGrad_LargeSizeTensor) {
   TestLayerNormGrad(X_dims, -1, 5e-3);
 }
 
+static void TestInvertibleLayerNormGrad(
+    const std::vector<int64_t>& x_dims,
+    int64_t axis = -1,
+    double error_tolerance = 1e-4) {
+  const std::vector<int64_t>& n_x_m_dims = x_dims;
+  std::vector<int64_t> n_dims, m_dims;
+  ASSERT_TRUE(SplitDims(n_x_m_dims, axis, n_dims, m_dims).IsOK());
+
+  const auto N = std::accumulate(n_dims.begin(), n_dims.end(), static_cast<int64_t>(1), std::multiplies<>{});
+  const auto M = std::accumulate(m_dims.begin(), m_dims.end(), static_cast<int64_t>(1), std::multiplies<>{});
+
+  CompareOpTester test{"InvertibleLayerNormalizationGrad", 1, kMSDomain};
+
+  test.AddAttribute("axis", axis);
+
+  RandomValueGenerator random{};
+  const auto Y_grad_data = random.Uniform<float>(n_x_m_dims, k_random_data_min, k_random_data_max);
```

Comment:
I didn't see any mixedprecision tests, might need to add them

Reply:
Agree. But current test framework such as implementation of CompareWithCPU needs to be changed to support fp16. It is better to have a separate PR for it.